wicket-core/src/main/java/org/apache/wicket/markup/html/image/ExternalImage.java

```diff
@@ -0,0 +1,317 @@
+	{
+		this(id, srcModel, Model.ofList(Collections.<Serializable> emptyList()));
+	}
+
+	/**
+	 * Creates an external image
+	 * 
+	 * @param id
+	 *            the component id
+	 * @param srcModel
+	 *            the model source URL
+	 * @param srcSetModel
+	 *            a model list of URLs placed in the srcSet attribute
+	 */
+	public ExternalImage(String id, IModel<Serializable> srcModel,
+		IModel<List<Serializable>> srcSetModel)
+	{
+		super(id, srcModel);
+		this.srcSetModel = srcSetModel;
+	}
+
+	@Override
+	protected void onComponentTag(ComponentTag tag)
+	{
+		super.onComponentTag(tag);
+
+		if ("source".equals(tag.getName()))
+		{
+			buildSrcSetAttribute(tag, getSrcSetModel());
+		}
+		else
+		{
+			checkComponentTag(tag, "img");
+			buildSrcAttribute(tag, getDefaultModel());
+			buildSrcSetAttribute(tag, getSrcSetModel());
+		}
+
+		buildSizesAttribute(tag);
+
+		Cors crossOrigin = getCrossOrigin();
+		if (crossOrigin != null && Cors.NO_CORS != crossOrigin)
+		{
+			tag.put("crossOrigin", crossOrigin.getRealName());
+		}
+	}
+
+	/**
+	 * Builds the src attribute
+	 *
+	 * @param tag
+	 *            the component tag
+	 * @param srcModel
+	 *            the model containing the src URL
+	 */
+	protected void buildSrcAttribute(final ComponentTag tag, IModel<?> srcModel)
+	{
+		tag.put("src", String.valueOf(srcModel.getObject()));
+	}
+
+	/**
+	 * Builds the srcset attribute if multiple models are found as varargs
+	 *
+	 * @param tag
+	 *            the component tag
+	 * @param srcSetModel
+	 *            the models containing the src set URLs
+	 */
+	protected void buildSrcSetAttribute(final ComponentTag tag,
+		IModel<List<Serializable>> srcSetModel)
+	{
+		int srcSetPosition = 0;
+		List<Serializable> srcSetItems = srcSetModel.getObject();
+		for (Serializable srcSet : srcSetItems)
+		{
+			String srcset = tag.getAttribute("srcset");
+			String xValue = "";
+
+			// If there are xValues set process them in the applied order to the srcset
+			// attribute.
+			if (xValues != null)
+			{
+				xValue = xValues.size() > srcSetPosition && xValues.get(srcSetPosition) != null
+					? " " + xValues.get(srcSetPosition) : "";
+			}
+			tag.put("srcset", (srcset != null ? srcset + ", " : "") + srcSet + xValue);
+			srcSetPosition++;
+		}
+	}
+
+	/**
+	 * builds the sizes attribute of the img tag
+	 *
+	 * @param tag
+	 *            the component tag
+	 */
+	protected void buildSizesAttribute(final ComponentTag tag)
+	{
+		// if no sizes have been set then don't build the attribute
+		if (sizes == null)
+		{
+			return;
+		}
+		String sizes = "";
+		for (String size : this.sizes)
+		{
+			sizes += size + ",";
+		}
+		int lastIndexOf = sizes.lastIndexOf(",");
+		if (lastIndexOf != -1)
+		{
+			sizes = sizes.substring(0, lastIndexOf);
+		}
+		if (!"".equals(sizes))
+		{
+			tag.put("sizes", sizes);
+		}
+	}
+
+	/**
+	 * @param values
+	 *            the x values to be used in the srcset
+	 */
+	public void setXValues(String... values)
```

Comment:
Good idea - looks much better then setXValues(null) - I also improve Image / Source with that call.